lib/resources/users.js

```diff
@@ -31,52 +34,81 @@ module.exports = (service, endpoint) => {
       ])
         .then(([ exact, list ]) => exact.map((x) => [ x ]).orElse(list.orElse([]))))));
 
-  service.post('/users', endpoint(({ Users, mail }, { body, auth }) =>
-    auth.canOrReject('user.create', User.species)
-      .then(() => User.fromApi(body).forV1OnlyCopyEmailToDisplayName())
-      .then(Users.create)
-      .then((savedUser) => (isPresent(body.password)
-        ? Users.updatePassword(savedUser, body.password)
-          .then(() => mail(savedUser.email, 'accountCreatedWithPassword'))
-        : Users.provisionPasswordResetToken(savedUser)
-          .then((token) => mail(savedUser.email, 'accountCreated', { token })))
-        .then(always(savedUser)))));
+  if(oidc.isEnabled()) {
+    // Same as non-OIDC, except that password is random & unguessable
+    service.post('/users', endpoint(({ Users, mail }, { body, auth }) =>
+      auth.canOrReject('user.create', User.species)
+        .then(() => User.fromApi(body).forV1OnlyCopyEmailToDisplayName())
+        .then(Users.create)
+        // REVIEW: is this considered an acceptable way of blocking password-based login?
+        .then((savedUser) => Users.updatePassword(savedUser, randomPassword())
```

Comment:
@ktuite brought up an interesting case during the meeting yesterday, of what happens if an existing deployment wants to start using SSO. I see that this endpoint would prevent password-based login for users created after the move to SSO, but should we also try to prevent password-based login for users created before the move? Is that already handled somehow?

Reply:
At this stage, enabling OIDC is all-in, so it will need to mentioned in documentation rather than handled in code.

Reply:
I guess a server admin could use the Central CLI to give any existing users random passwords so that they can't use their old passwords. I can't think of a better way to automate the changing of passwords:

- Users could be deleted and recreated, but they wouldn't be linked to their previous activity.
- A password reset could be requested for every user, but that would send an email to everyone. (And maybe that endpoint just shouldn't exist anymore?)

Is using the CLI the strategy you recommend we mention in the docs? What do you mean that enabling OIDC is all-in?

Reply:
> A password reset could be requested for every user

Ah that endpoint doesn't exist anymore (which makes a lot of sense), so that wouldn't even be an option.